oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -202,6 +250,19 @@ public void createAssertion_correct() throws IOException {
     assertEquals(Joiner.on(' ').join(scopes), payload.get("scope"));
   }
 
+  @Test
+  public void createAssertion_custom_lifetime() throws IOException {
+    ServiceAccountCredentials credentials = createDefaultBuilder().setLifetime(4000).build();
+
+    JsonFactory jsonFactory = OAuth2Utils.JSON_FACTORY;
+    long currentTimeMillis = Clock.SYSTEM.currentTimeMillis();
+    String assertion = credentials.createAssertion(jsonFactory, currentTimeMillis, null);
+
+    JsonWebSignature signature = JsonWebSignature.parse(jsonFactory, assertion);
+    JsonWebToken.Payload payload = signature.getPayload();
+    assertEquals(currentTimeMillis / 1000 + 4000, (long) payload.getExpirationTimeSeconds());
```

Comment:
This could possibly fail (flakily) if the processing crosses the second barrier.

A few options:
* assert this within 1 second difference
* inject a Clock into the method (that defaults to the system clock) for testing (example: [Builder](https://github.com/googleapis/google-auth-library-java/blob/72a00b9390191da669f58440531671c4b21a71b8/oauth2_http/java/com/google/auth/oauth2/JwtCredentials.java#L249-L252), [constructor](https://github.com/googleapis/google-auth-library-java/blob/72a00b9390191da669f58440531671c4b21a71b8/oauth2_http/java/com/google/auth/oauth2/JwtCredentials.java#L93))

Reply:
This looks fine. Clock is only used when we create the `currentTimeMills` value. The `createAssertion` code and the assertion code use the same `currentTimeMills` value, and clock is not involved there. 

```
    long currentTimeMillis = Clock.SYSTEM.currentTimeMillis();
    String assertion = credentials.createAssertion(jsonFactory, currentTimeMillis, null);

    JsonWebSignature signature = JsonWebSignature.parse(jsonFactory, assertion);
    JsonWebToken.Payload payload = signature.getPayload();
    assertEquals(currentTimeMillis / 1000 + 4000, (long) payload.getExpirationTimeSeconds());
```